benefits/enrollment/views.py

```diff
@@ -88,6 +90,13 @@ def index(request):
             if e.response.status_code == 409:
                 analytics.returned_success(request, eligibility.group_id)
                 return success(request)
+            elif e.response.status_code >= 500:
+                analytics.returned_error(request, str(e))
+                sentry_sdk.capture_exception(e)
+
+                # overwrite origin so that CTA takes user to agency index
+                session.update(request, origin=agency.index_url)
+                return system_error(request)
```

Comment:
I wonder if it makes sense to do the analytics, sentry capture, and the session update inside the `system_error` function? Otherwise I imagine this will need to be repeated every time this system error is called?

Reply:
Yeah I think it would make sense to do that. I was hesitating on it though because I wasn't sure how we would pass arguments to the view function when in later PRs we'll need to call `reverse` on this view function and/or call it from Javascript in the front-end.

I'm seeing now that passing arguments in those situations [isn't too bad](https://docs.djangoproject.com/en/5.0/topics/http/urls/#reverse-resolution-of-urls) assuming that we're ok with having captured values in the URL, e.g. `/system_error/<str:error>/<agency:agency>`...? 

This approach feels weird to me, maybe because our code doesn't normally pass things in the URL just for analytics / Sentry. Any thoughts or alternatives I'm not seeing?

Reply:
Ah, no that's a good point -- I think we could get away with if it was just the agency, grabbing that out of the session. But I overlooked the exception, and yeah I agree sending that in the URL from Javascript is weird.

Reply:
Good idea, thanks! I added a test assertion for the session update in https://github.com/cal-itp/benefits/pull/2088/commits/87e4dfca1554c12af0832a9b0064ff4fbbede3bd and moved the session update into the view function in https://github.com/cal-itp/benefits/pull/2088/commits/8d551832a95563129271720fba747e6907a16cdd